### PR TITLE
Modernize qualtrics-survey xblock

### PR DESCRIPTION
**Pull request by stvstnfrd**
following changes made to image-modal xblock

### CODE REVIEW

**Comment by stvstnfrd** [comment]
The `README` still needs a little updating, but everything else should be good to go.

Test with:
- `make`
- `make test && make runserver`

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+|image-lms-view-normal|
+
+
+Installation, system administrator
+----------------------------------
+
+To install the XBlock on your platform,
+add the following to your `requirements.txt` file:
+
+    xblock-qualtrics-survey
```
Comment (stvstnfrd) [inline README.rst:25]: TODO: Publish this to PyPI.

**Review by stvstnfrd — COMMENTED** [review]
I'll tweak the README a little prior to merge (mostly just needs screenshots and walkthrough videos), but the code is ready and contains the latest updates from the `image-modal`.

@caesar2164 @dcadams Please review.

**Comment by stvstnfrd** [comment]
Actually, I take that back; I missed a change or two; please hold off.

**Comment by stvstnfrd** [comment]
@caesar2164 @dcadams Okay, ready for review, for real ;)

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,7 @@
 *.egg-info/
 node_modules/
 *.log
+package-lock.json
```
Comment (caesar2164) [inline .gitignore:4]: Do we need to add `.dockerignore` here as well?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+FROM stvstnfrd/workbench:latest
```
Comment (caesar2164) [inline Dockerfile:1]: Shouldn't this be changed for our master branch?

File: Makefile [hunk 4]
```diff
@@ -1,61 +1,89 @@
...
+    po_files = $(translation_root)/en/LC_MESSAGES/django.po
+endif
 
 .PHONY: help
 help:  ## This.
 	@perl -ne 'print if /^[a-zA-Z_-]+:.*## .*$$/' $(MAKEFILE_LIST) \
-	| sort \
 	| awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}'
 
+.PHONY: runserver
```
Comment (caesar2164) [inline Makefile:22]: works!

File: Makefile [hunk 5]
```diff
@@ -1,61 +1,89 @@
...
+runserver: build_docker  ## Run server inside XBlock Workbench container
+	$(docker_run) $(_NAME)
+
 .PHONY: clean
```
Comment (caesar2164) [inline Makefile:26]: works!

File: Makefile [hunk 6]
```diff
@@ -1,61 +1,89 @@
...
-	pip install -e .
-	cd $(sdk_root) && python manage.py migrate
-	cd $(sdk_root) && python manage.py runserver 0.0.0.0:8000
+.PHONY: static
+static: requirements_js $(css_files)  ## Compile the less->css
+$(module_root)/public/%.css: $(module_root)/public/%.less
+	@echo "$< -> $@"
+	node_modules/less/bin/lessc $< $@
 
 .PHONY: test
```
Comment (caesar2164) [inline Makefile:52]: works!

File: qualtricssurvey/public/view.css [hunk 7]
```diff
@@ -1,5 +1,5 @@
 .qualtricssurvey_block .qualtrics-button,
 .qualtricssurvey_block .qualtrics-button:hover,
 .qualtricssurvey_block .qualtrics-button:visited {
-  color: white;
+  /* color: white;*/
```
Comment (caesar2164) [inline qualtricssurvey/public/view.css]: why not just deleting this?

File: qualtricssurvey/public/view.less [hunk 8]
```diff
@@ -1,5 +1,7 @@
 .qualtricssurvey_block {
-    .qualtrics-button, .qualtrics-button:hover, .qualtrics-button:visited {
-        color: white;
+    .qualtrics-button,
+    .qualtrics-button:hover,
+    .qualtrics-button:visited {
+        /* color: white;*/
```
Comment (caesar2164) [inline qualtricssurvey/public/view.less]: again, why commenting this line?

File: qualtricssurvey/templates/view.html [hunk 9]
```diff
@@ -0,0 +1,9 @@
+<div class="qualtricssurvey_block">
+    <p>{{message}}</p>
+    <p>
+        <a class="button primary-button qualtrics-button"
+           href="https://{{your_university}}.qualtrics.com/jfe/form/{{survey_id}}{{user_id_string}}"
+           target="_blank"
+        >{{link_text}}</a>
```
Comment (caesar2164) [inline qualtricssurvey/templates/view.html]: Closing the `a` tag on this line makes it look weird. Any other way we can do this?

**Review by caesar2164 — CHANGES_REQUESTED** [review]
First pass with obvious stuff.

File: .gitignore (hunk 2 above)
Comment (stvstnfrd) [reply .gitignore:4]: No, I'm not sure why we'd want to ignore changes to the `.dockerignore` file...?

File: qualtricssurvey/public/view.css (hunk 7 above)
Comment (stvstnfrd) [reply qualtricssurvey/public/view.css]: This is the compiled file, so I'll resolve this thread and respond to the `.less` file.

File: qualtricssurvey/templates/view.html (hunk 9 above)
Comment (stvstnfrd) [reply qualtricssurvey/templates/view.html]: Eh, we're actually closing it on the same line that we're opening it.

Splitting it up maybe looks weirder?
```
        <a
          class="button primary-button qualtrics-button"
          href="https://{{your_university}}.qualtrics.com/jfe/form/{{survey_id}}{{user_id_string}}"
          target="_blank"
        >
          {{link_text}}
        </a>
```

Does that work?

File: Dockerfile (hunk 3 above)
Comment (stvstnfrd) [reply Dockerfile:1]: This is just a stop-gap solution.

Ultimately, we'll use a container image from edx,
but we're still waiting for a few more changes to land upstream.

File: qualtricssurvey/public/view.less (hunk 8 above)
Comment (stvstnfrd) [reply qualtricssurvey/public/view.less]: Removed in new commit.

File: .gitignore (hunk 2 above)
Comment (caesar2164) [reply .gitignore:4]: I guess that's true. I was just thinking that dot files are usually not for git to look at...

File: qualtricssurvey/public/view.less (hunk 8 above)
Comment (caesar2164) [reply qualtricssurvey/public/view.less]: thanks.

File: qualtricssurvey/templates/view.html (hunk 9 above)
Comment (caesar2164) [reply qualtricssurvey/templates/view.html]: Yes, I like that better. that makes it formatted just like tags that don't have lots of attributes.

**Review by caesar2164 — DISMISSED** [review]
Nice, thanks for the changes!

**Comment by stvstnfrd** [comment]
@caesar2164 README updated and commits squashed.

@mondiaz The documentation can be previewed here:
- https://github.com/Stanford-Online/xblock-qualtrics-survey/blob/modernize/README.rst

**Comment by mondiaz** [comment]
@stvstnfrd Thanks for tagging me. I checked the documentation you linked to, and had a few notes:

1. The image for image-cms-add is broken (just displays a broken image icon).
2. For the "a" parameter name for userid to work, we (Kathy, really) set up embedded data in the Qualtrics surveys to pass that value through to the Qualtrics survey reports. 
3. We don't seem to need to add any name to the Advanced Module List for the qualtrics survey xblock to work anymore. 

Screen Shot 2019-07-09 at 3 04 33 PM

**Comment by stvstnfrd** [comment]
@mondiaz 

> 1. The image for image-cms-add is broken (just displays a broken image icon).

Fixed! Nice catch.

 
> 3. We don't seem to need to add any name to the Advanced Module List for the qualtrics survey xblock to work anymore.

Yeah, that step is optional on Lagunita, but required on any other instance (edx.org, opencraft, etc.).
We have an extra setting `XBLOCKS_ALWAYS_IN_STUDIO` that allows us to omit that for common XBlocks on our site, but I want the docs to be thorough for community members. It shouldn't hurt to add though :)


> 2. For the "a" parameter name for userid to work, we (Kathy, really) set up embedded data in the Qualtrics surveys to pass that value through to the Qualtrics survey reports.

Can we chat F2F about this?

**Comment by stvstnfrd** [comment]
@mondiaz @caesar2164 Changes pushed.

**Comment by mondiaz** [comment]
@stvstnfrd 👍 lgtm, thanks!

**Review by caesar2164 — APPROVED** [review]

**Pull request merged**